modules/gapi/include/opencv2/gapi/util/variant.hpp

```diff
@@ -428,7 +506,72 @@ namespace util
     {
         return !(lhs == rhs);
     }
-} // namespace cv
+
+namespace detail
+{
+    template<std::size_t CurIndex, std::size_t ElemCount,
+             typename Visitor, typename Variant, typename... VisitorArgs>
+    typename Visitor::result_type apply_visitor_impl(Visitor&& visitor,
+                                                     Variant& v,
+                                                     std::true_type processed,
+                                                     VisitorArgs&& ...args)
+    {
+        suppress_unused_warning(visitor);
+        std::array<bool, sizeof...(VisitorArgs)> dummy{(suppress_unused_warning(args),true)...};
+        suppress_unused_warning(dummy);
+        suppress_unused_warning(v);
+        suppress_unused_warning(processed);
+        return {};
+    }
+
+    template<std::size_t CurIndex, std::size_t ElemCount,
+             typename Visitor, typename Variant, typename... VisitorArgs>
+    typename Visitor::result_type apply_visitor_impl(Visitor&& visitor,
+                                                     Variant&& v,
+                                                     std::false_type not_processed,
+                                                     VisitorArgs&& ...args)
+    {
+        suppress_unused_warning(not_processed);
+        if(v.index() == CurIndex)
+        {
+            return visitor.operator() (CurIndex, get<CurIndex>(v), std::forward<VisitorArgs>(args)... );
```

Comment:
so that's it, we have an early return here on the match. Y do we need this `not_processed` thing then?